enhancements/ingress/set-delete-http-headers.md

```diff
@@ -0,0 +1,1031 @@
+  vars set in the router deployment using `oc -n openshift-ingress get deployment/<router-name> -o yaml`
+- If the router pods are in `CrashLoopBackOff` state after setting the `httpHeaders` in the IngressController Spec then check the router pod logs using `oc logs <router pod name> -n openshift-ingress` and also check the ingress-operator pod logs .
+- If the router pods are in `CrashLoopBackOff` state after setting the `httpHeaders` in the Route Spec then check the router pod logs using `oc logs <router pod name> -n openshift-ingress`.
+- In case of second and third point you may also run `oc debug deployment router-name -n openshift-ingress`, then check as per the following:
+  The reason to use oc debug is that `oc rsh` or `oc cp` might not work when the pod is in `CrashLoopBackOff` state.
+  - If HTTP headers were set/delete via IngressController Spec then you need check the front end sections for `http-response/http-request set-header` or `http-response/http-request del-header` stanzas set in the `haproxy.config`. Check
+    if the desired value was set.
+  - If HTTP headers were set/delete via RouteSpec then you need to check the Plain backend section for `http-response/http-request set-header` or `http-response/http-request  del-header` stanzas set in the `haproxy.config`. Check if the desired
+    value was set.
+- If the API fields are not getting set via IngressController Spec or Route Spec i.e. if `oc patch` is failing to edit values of the Ingress Controller or Route CR 
+  then check the openshift-apiserver logs. You can even run `oc explain route.spec.httpHeaders` or `oc explain ingresscontroller.spec.httpHeaders` to understand how to set the `httpHeaders` field.
+- Check [access logs](https://docs.openshift.com/container-platform/4.12/rest_api/operator_apis/ingresscontroller-operator-openshift-io-v1.html#spec-logging-access) of the haproxy if you want to check what HTTP Response or Request was set for the frontend and backend.
+
+## Implementation History
+
+## Alternatives
+### Avoid Using Union Types
+Using union types makes the newly added API fields more verbose and deeply nested than is strictly necessary. The API could be designed to avoid the use of unions
+by changing the `IngressControllerHTTPHeader` and `RouteHTTPHeaders` types to flat structures, using CEL to require or prohibit specifying a header value based on the specified action.
+In this alternative, `IngressControllerHTTPHeader` would look like the following (some extraneous details are omitted):
+```go
+type IngressControllerHTTPHeader struct {
+    Name string `json:"name"`
+    // +kubebuilder:validation:Enum:=Set;Delete
+    // Action IngressControllerHTTPHeaderAction `json:"action,omitempty"`
+    // +kubebuilder:validation:XValidation:rule="has(self.value) && self.action == 'Delete'", message="Value is not required while performing delete action."
```

Comment:
Nit - this doesn't line up with the updated API, which has Action rather than Value, and does use a union type.